packages/garbo/src/resources/mayam.ts

```diff
@@ -0,0 +1,262 @@
+/* eslint-disable libram/verify-constants */
+import { cliExecute, Item, myLevel, useFamiliar } from "kolmafia";
+import { $item, clamp, get, maxBy, MayamCalendar } from "libram";
+import { garboValue } from "../garboValue";
+import { GarboTask } from "../tasks/engine";
+import getExperienceFamiliars from "../familiar/experienceFamiliars";
+import { globalOptions } from "../config";
+
+type MayamRingSummon = {
+  choice: string;
+  value: () => number;
+  ring: number;
+};
+
+type Resonance = {
+  rings: string;
+  item: Item;
+};
+
+const RESONANCE: Resonance[] = [
+  { rings: "eye yam2 eyepatch yam4", item: $item`Mayam spinach` },
+  { rings: "vessel yam2 cheese explosion", item: $item`stuffed yam stinkbomb` },
+  { rings: "yam1 meat cheese yam4", item: $item`yam and swiss` },
+  { rings: "yam1 lightning yam3 clock", item: $item`yam battery` },
+  { rings: "yam1 yam2 yam3 explosion", item: $item`thanksgiving bomb` },
+  // {rings: "yam1 meat eyepatch yam4", item: $item`yamtility belt`}
+];
+
+const yamUses = () => get("_mayamSymbolsUsed");
+
+function findResonance(rings: string): Resonance | undefined {
+  return RESONANCE.find((resonance) => resonance.rings === rings);
+}
+
+// Function to find the resonance value
+export function findResonanceValue(input: string): number {
+  const result = findResonance(input);
+
+  // Check if the result is found
+  if (result) {
+    return garboValue(result.item);
+  }
+
+  return 0;
+}
+
+const MAYAM_RING_OPTIONS: MayamRingSummon[] = [
+  {
+    choice: "yam1",
+    value: () => garboValue($item`yam`),
+    ring: 1,
+  },
+  {
+    choice: "yam2",
+    value: () => garboValue($item`yam`),
+    ring: 2,
+  },
+  {
+    choice: "yam3",
+    value: () => garboValue($item`yam`),
+    ring: 3,
+  },
+  {
+    choice: "yam4",
+    value: () => garboValue($item`yam`),
+    ring: 4,
+  },
+  {
+    choice: "sword",
+    value: () => 0,
+    ring: 1,
+  },
+  {
+    choice: "eye",
+    value: () => 0, // +30% item drop, I don't know how to value that?
+    ring: 1,
+  },
+  {
+    choice: "chair",
+    value: () => 0, // +5 free rests, I don't know how to value that?
+    ring: 1,
+  },
+  {
+    choice: "fur",
+    value: () =>
+      Math.max(
+        0,
+        ...getExperienceFamiliars("free").map(
+          ({ expectedValue }) => expectedValue / 12,
+        ),
+      ) * 100,
+    ring: 1,
+  },
+  {
+    choice: "vessel",
+    value: () => 0, // How do we value MP regen?
+    ring: 1,
+  },
+  {
+    choice: "lightning",
+    value: () => 0,
+    ring: 2,
+  },
+  {
+    choice: "bottle",
+    value: () => 0, // How do we value HP regen?
+    ring: 2,
+  },
+  {
+    choice: "meat",
+    value: () => clamp(myLevel() * 100, 100, 1500),
+    ring: 2,
```

Comment:
We shouldn't need to store that, we can access it by looking at the `choice` parameter. Alternately, I'd break this up into either an object containing four arrays, or just four separate arrays, one for each ring.

Reply:
This comment is the key to your most recent issue: you either want to structure it as like ```
const MAYAM_RING_OPTIONS = {
  1: [
      { choice: " vessel", value: () => 0 },
      etc
    ],
    2: [ etc ]
    etc
  } as const;
  ```
  or just as four separate arrays. But I'd do the former.

Reply:
the `as const` will help typescript handle the types, at least a little

Reply:
Four separate arrays was easier for me to understand and manage.